klayout/drc/run_drc.py

```diff
@@ -0,0 +1,709 @@
+    run_drc.py (--path=<file_path>) (--variant=<combined_options>) [--verbose] [--table=<table_name>]... [--mp=<num_cores>] [--run_dir=<run_dir_path>] [--topcell=<topcell_name>] [--thr=<thr>] [--run_mode=<run_mode>] [--no_feol] [--no_beol] [--connectivity] [--density] [--density_only] [--antenna] [--antenna_only] [--no_offgrid]
+
+Options:
+    --help -h                           Print this help message.
+    --path=<file_path>                  The input GDS file path.
+    --variant=<combined_options>        Select combined options of metal_top, mim_option, and metal_level. Allowed values (A, B, C).
+                                        variant=A: Select  metal_top=30K  mim_option=A  metal_level=3LM
+                                        variant=B: Select  metal_top=11K  mim_option=B  metal_level=4LM
+                                        variant=C: Select  metal_top=9K   mim_option=B  metal_level=5LM
+    --topcell=<topcell_name>            Topcell name to use.
+    --table=<table_name>                Table name to use to run the rule deck.
+    --mp=<num_cores>                    Run the rule deck in parts in parallel to speed up the run. [default: 1]
+    --run_dir=<run_dir_path>            Run directory to save all the results [default: pwd]
+    --thr=<thr>                         The number of threads used in run.
+    --run_mode=<run_mode>               Select klayout mode Allowed modes (flat , deep, tiling). [default: flat]
+    --no_feol                           Turn off FEOL rules from running.
+    --no_beol                           Turn off BEOL rules from running.
+    --connectivity                      Turn on connectivity rules.
+    --density                           Turn on Density rules.
+    --density_only                      Turn on Density rules only.
+    --antenna                           Turn on Antenna checks.
+    --antenna_only                      Turn on Antenna checks only.
+    --no_offgrid                        Turn off OFFGRID checking rules.
+    --verbose                           Detailed rule execution log for debugging.
+"""
+
+from docopt import docopt
+import os
+import xml.etree.ElementTree as ET
+import logging
+import klayout.db
+import glob
+from datetime import datetime
+from subprocess import check_call
+import shutil
+import concurrent.futures
+import traceback
+
+
+def get_rules_with_violations(results_database):
+    """
+    This function will find all the rules that has violated in a database.
+
+    Parameters
+    ----------
+    results_database : string or Path object
+        Path string to the results file
+
+    Returns
+    -------
+    set
+        A set that contains all rules in the database with violations
+    """
+
+    mytree = ET.parse(results_database)
+    myroot = mytree.getroot()
+
+    all_violating_rules = set()
+
+    for z in myroot[7]:  # myroot[7] : List rules with viloations
+        all_violating_rules.add(f"{z[1].text}".replace("'", ""))
```

Comment:
curious if @mithro has a preference for `""` vs `''`, see https://peps.python.org/pep-0008/#string-quotes
(I usually use single ones)

Reply:
The violation name comes with single quote. FYI

Reply:
you mean with flake8?